File: mobile/shared/src/commonMain/kotlin/com/unibank/shared/domain/model/Loan.kt
package com.unibank.shared.domain.model

data class LoanApplicationResult(
    val success: Boolean,
    val message: String,
    val loanId: String,
    val reference: String,
    val status: LoanStatus,
    val principal: Money,
    val interestRate: String,
    val monthlyPayment: Money,
    val tenureMonths: Int,
    val creditScore: Int,
    val newBalance: Money,
)

data class LoanDetail(
    val loanId: String,
    val reference: String,
    val status: LoanStatus,
    val principal: Money,
    val outstandingBalance: Money,
    val interestRate: String,
    val tenureMonths: Int,
    val monthlyPayment: Money,
    val purpose: String,
    val paymentsMade: Int,
    val totalPayments: Int,
    val nextPaymentDate: String,
    val createdAt: String,
    val creditScore: Int,
)

data class LoanSummary(
    val loanId: String,
    val reference: String,
    val status: LoanStatus,
    val principal: Money,
    val outstandingBalance: Money,
    val monthlyPayment: Money,
    val paymentsMade: Int,
    val totalPayments: Int,
    val createdAt: String,
)

data class LoanScheduleEntry(
    val paymentNumber: Int,
    val principalAmount: Money,
    val interestAmount: Money,
    val totalPayment: Money,
    val remainingBalance: Money,
    val dueDate: String,
    val isPaid: Boolean,
)

enum class LoanStatus {
    UNSPECIFIED, PENDING, APPROVED, REJECTED, DISBURSED, REPAYING, PAID_OFF, DEFAULTED,
}

File: server/UniBank.Core/Modules/Loans/Domain/Entities/Loan.cs
using UniBank.SharedKernel.Domain;

namespace UniBank.Core.Modules.Loans.Domain.Entities;

/// <summary>
/// Represents a personal unsecured loan application and its lifecycle.
/// Tracks principal, outstanding balance, repayment schedule, and credit scoring.
/// </summary>
public sealed class Loan : AggregateRoot
{
    public Guid AccountId { get; set; }
    public decimal Principal { get; set; }
    public decimal OutstandingBalance { get; set; }
    public decimal InterestRate { get; set; }
    public int TenureMonths { get; set; }
    public decimal MonthlyPayment { get; set; }
    public string Purpose { get; set; } = default!;
    public string Status { get; set; } = "pending"; // pending, approved, rejected, disbursed, repaying, paid_off, defaulted
    public int CreditScore { get; set; }
    public int PaymentsMade { get; set; }
    public string Reference { get; set; } = default!;
    public string Currency { get; set; } = "ZWG";
    public string TenantId { get; set; } = default!;
    public DateTime? DisbursedAt { get; set; }
    public DateTime? CompletedAt { get; set; }
    public DateTime? DeletedAt { get; set; }

    public ICollection<LoanPayment> Payments { get; set; } = [];
}
